### ISSUE TITLE
Mann-Whitney-U Test producing strange results

### ISSUE THREAD

**Issue by hkoestin** [accord-net/framework#857]
Dear @cesarsouza 

I just found out, that the [MannWhitneyWilcoxonTest ](https://github.com/accord-net/framework/blob/master/Sources/Accord.Statistics/Testing/TwoSample/MannWhitneyWilcoxonTest.cs)class may be buggy.

Given the following sample vector, when I pass it in as sample one and sample two vector, the Accord.NET implementation shows a p-Value of 0.41 or the like.
According to R, for the given vector, the p-Value should be 1 (exactly).

```R
a <- c(250,200,450,400,250,250,350,0,200,400,300,600,200,200,
       550,100,300,250,350,200,550,200,450,400,200,400,450,
       200,400,400,500,450,300,250,200)
b <- c(250,200,450,400,250,250,350,0,200,400,300,600,200,200,
       550,100,300,250,350,200,550,200,450,400,200,400,450,
       200,400,400,500,450,300,250,200)

wilcox.test(a, b)
```
Gives the following output:

```R
	Wilcoxon rank sum test with continuity correction

data:  a and b
W = 612.5, p-value = 1
alternative hypothesis: true location shift is not equal to 0

Warning message:
In wilcox.test.default(a, b) : cannot compute exact p-value with ties
```

I have other sample vectors, which show the same behavior. R always shows p-Value of 1 for them.

Thanks in advance,
Harald

**Comment by hkoestin** [comment]
I searched the code base a little. 
Found the following "issues":
- the parameter **adjustForTies** is never actually used: https://github.com/accord-net/framework/blob/98e6d94b43472ed1277f54695eb586ef80b9432c/Sources/Accord.Statistics/Testing/TwoSample/MannWhitneyWilcoxonTest.cs#L182
- the calculation of "n" is not needed anymore (it was needed, when the Debug.Assert from line 209 was still in): https://github.com/accord-net/framework/blob/98e6d94b43472ed1277f54695eb586ef80b9432c/Sources/Accord.Statistics/Testing/TwoSample/MannWhitneyWilcoxonTest.cs#L186

**Comment by cesarsouza** [comment]
Hi @hkoestin,

Many thanks for opening the issue and for the observations regarding the unused variables. I will be committing a fix for those and for the original issue in the next minutes. 

Regards,
Cesar

**Comment by cesarsouza** [comment]
Hi @hkoestin,

I am uploading new pre-release packages (3.7.1-alpha) to NuGet with the latest corrections. They should be available soon.

Regards,
Cesar

**Comment by hkoestin** [comment]
Hi @cesarsouza 

thanks for committing the fix and providing the new alpha version.
I just tested it, and the results seem to be much closer to R than before.

So: works fine for me now! Thank you!

Regards
Harald

**Comment by hkoestin** [comment]
Oops. Pressed the wrong button

**Comment by cesarsouza** [comment]
Fixed in 3.8.0.
